thunder/__init__.py

```diff
@@ -589,39 +589,39 @@ def get_computation_and_inputs(*args, **kwargs):
                     # by split_forward_backward
                     extraces = cs.last_traces
 
-            if backward_trc is None:
```

Comment:
This deletes too much. Only the for loop L593-605 should go

Reply:
Ok! Sounds good. Let me update that!